[Admin] Cart promotions translations for labels

ernestWarwas opened the pull request:
| Q               | A                                                            |
|-----------------|--------------------------------------------------------------|
| Branch?         | 1.13 <!-- see the comment below -->                  |
| Bug fix?        | no                                                       |
| New feature?    | yes                                                       |
| BC breaks?      | no                                                       |
| Deprecations?   | no |
| License         | MIT                                                          |

This PR provides the ability to translate the labels of cart promotions and the second part of showing them in shop ui and admin panel in orders will be added in other pr.

<img width="1438" alt="image" src="https://user-images.githubusercontent.com/28228691/192985913-b9a3741e-5941-439b-95bd-e4741c7b9619.png">


<!--
 - Bug fixes must be submitted against the 1.11 branch
 - Features and deprecations must be submitted against the 1.12 branch
 - Make sure that the correct base branch is set

 To be sure you are not breaking any Backward Compatibilities, check the documentation:
 https://docs.sylius.com/en/latest/book/organization/backward-compatibility-promise.html
-->

ernestWarwas committed:
[Behat][Ui] Ui implementation

ernestWarwas committed:
[Spec] Changes in spec

[Spec] Changes in spec

fix behat scenario

ernestWarwas committed:
[API] contract tests

[API] contract tests

[API] contract tests

[Behat] scenario improvements

license block

GSadee commented on features/promotion/managing_promotions/adding_promotion_in_different_languages.feature:
```diff
@@ -0,0 +1,21 @@
+        And that channel allows to shop using "English (United States)" and "Polish (Poland)" locales
+        And I am logged in as an administrator
+
+    @ui @javascript @no-api
```
Why did you add here `@no-api` tag?
```suggestion
    @ui @javascript
```

GSadee commented on src/Sylius/Behat/Context/Ui/Admin/ManagingPromotionsContext.php:
```diff
@@ -107,6 +109,23 @@ public function iAddIt()
+     */
+    public function iSpecifyItsLabelInLocaleCode(string $label, string $localeCode): void
+    {
+        $this->createPage->specifyLabel($label, $this->localeConverter->convertNameToCode($localeCode));
```
As this is already a locale code, why are you converting it to code? I can probably answer myself, if I see correctly, you have not imported LocaleContext transformer 🤔

GSadee commented on src/Sylius/Behat/Context/Ui/Admin/ManagingPromotionsContext.php:
```diff
@@ -107,6 +109,23 @@ public function iAddIt()
+    }
+
+    /**
+     * @When the :promotion promotion should have a label :label in :locale locale
```
```suggestion
     * @When the :promotion promotion should have a label :label in :localeCode locale
```
and the usage of `LocaleConverterInterface::convertNameToCode` should not be needed

GSadee commented on src/Sylius/Behat/Page/Admin/Promotion/CreatePage.php:
```diff
@@ -27,6 +27,13 @@ class CreatePage extends BaseCreatePage implements CreatePageInterface
     use NamesIt;
     use SpecifiesItsCode;
 
+    public function specifyLabel(string $label, string $locale): void
```
```suggestion
    public function specifyLabel(string $label, string $localeCode): void
```

GSadee commented on src/Sylius/Behat/Page/Admin/Promotion/CreatePage.php:
```diff
@@ -186,6 +193,18 @@ public function checkIfActionConfigurationFormIsVisible(): bool
         return $this->hasElement('amount');
     }
 
+    public function hasLabel(string $label, string $locale): bool
```
```suggestion
    public function hasLabel(string $label, string $localeCode): bool
```

GSadee commented on src/Sylius/Bundle/AdminBundle/Resources/views/Promotion/Show/_translations.html.twig:
```diff
@@ -0,0 +1,25 @@
+                <tr>
+                    <td class="three wide"><strong class="gray text">{{ 'sylius.ui.description'|trans }}</strong></td>
+                    <td>{{ translation.description|nl2br }}</td>
+                </tr>
```
Why did you also add a translatable description (as in the task there is only a requirement of adding a label)? And what is more, the promotion already has a description field, but not translatable.

ernestWarwas committed:
[Admin] remove description from translation as its not scope of task

[Admin] remove description from translation as its not scope of task

Renamed argument

GSadee commented on src/Sylius/Behat/Page/Admin/Promotion/CreatePage.php:
```diff
@@ -27,6 +27,13 @@ class CreatePage extends BaseCreatePage implements CreatePageInterface
 
+    public function specifyLabel(string $label, string $localeCode): void
+    {
+        $this->getDocument()->find('css', 'div[data-locale="' . $localeCode . '"]')->click();
```
As you are using `sprintf` method two lines below, you could use it also here to be consistent 💃🏻

GSadee commented on src/Sylius/Component/Promotion/spec/Model/PromotionSpec.php:
```diff
@@ -22,6 +22,12 @@
 
 final class PromotionSpec extends ObjectBehavior
 {
+    function let()
```
Missing return type

GSadee commented:
Thank you, Ernest! :1st_place_medal:
